src/EnergyPlus/HVACVariableRefrigerantFlow.cc

```diff
@@ -14429,12 +14442,20 @@ void VRFCondenserEquipment::VRFOU_CalcCompC(EnergyPlusData &state,
                     goto Label13;
                 }
 
-                if (CapDiff > (Tolerance * Cap_Eva0)) NumIteCcap = 999;
+                if (CapDiff > (Tolerance * Cap_Eva0)) {
+                    NumIteCcap = 999;
+                    CyclingRatio = (TU_load + Pipe_Q) * C_cap_operation / Cap_Eva1;
+                } else {
+                    CyclingRatio = 1.0;
+                }
 
-                Ncomp = this->RatedCompPower * CurveValue(state, this->OUCoolingPWRFT(CounterCompSpdTemp), T_discharge, T_suction);
+                Ncomp = this->RatedCompPower * CurveValue(state, this->OUCoolingPWRFT(CounterCompSpdTemp), T_discharge, T_suction) * CyclingRatio;
+                OUCondHeatRelease = Cap_Eva1 * CyclingRatio;
```

Comment:
I don't think this is correct. Earlier `OUCondHeatRelease = TU_load + Pipe_Q + Ncomp;`
Possibly `OUCondHeatRelease  *= CyclingRatio;` would be correct here?

Reply:
or would it be Ncomp + Cap_Eva1 * CyclingRatio? similar to the load > maximum capacity case
```
    if (CounterCompSpdTemp > NumOfCompSpdInput) {
        // Required load is beyond the maximum system capacity
        CompEvaporatingCAPSpd(NumOfCompSpdInput) =
            this->CoffEvapCap * this->RatedEvapCapacity * CurveValue(state, this->OUCoolingCAPFT(NumOfCompSpdInput), T_discharge, T_suction);
        OUCondHeatRelease = Ncomp + CompEvaporatingCAPSpd(NumOfCompSpdInput);
        CompSpdActual = this->CompressorSpeed(NumOfCompSpdInput);
        Ncomp = CompEvaporatingPWRSpd(NumOfCompSpdInput);
    }
```

Reply:
That sounds ok. This function is very hard to follow, so I'll trust you on that.

Reply:
It would be a nice test to plot some of the available report variables through a transition from CyclingRatio = 1 at some time step to CyclingRatio < 1 at future time steps to see if these report variables have a smooth transition from compressor on to compressor cycling. Whichever reports capture the equations being questioned here. Not sure why a component model is not reporting refrigerant pressures which could help in diagnostics. We should add refrigerant temperature/pressure reporting at some point.

Reply:
good idea. I will produce some plot showing how CyclingRatio transition from 1 to < 1 and the other way

Reply:
<img width="1030" alt="image" src="https://github.com/NREL/EnergyPlus/assets/13421161/c1281ed8-e5e0-4def-bfc6-c5c2a73a0f37">

[output change for cycling ratio from non-1 to 1.xlsx](https://github.com/NREL/EnergyPlus/files/14577311/output.change.for.cycling.ratio.from.non-1.to.1.xlsx)

I plotted two snapshot with cycling ratio changing from < 1 to 1 then back to < 1. The cooling electricity rate and condenser heat release seems to be smooth

Reply:
I'm curious why the VRF heat pump's cooling electricity rate is higher when the cycling ratio is less than 1 (for example, on 4/9 at 14:00 and 15:00) compared to when the cycling ratio is exactly 1 (on 4/9 at 16:00 and 17:00).

Reply:
That's a good question. From [this figure](https://github.com/NREL/EnergyPlus/pull/10353#issuecomment-2110831149) suction temperature is also changing which will change the refrigerant mass flow rate at the same volumetric flow rate. So there are other things changing in this model. I guess I would add suction temperature and refrigerant mass flow rate to these plots to see if something can be identified as the driving force for the difference in cooling electricity rate. I also see 2 variables for OU fan power. I'm not sure what N_fan_OU_evap and m_air_evap are regarding the outdoor unit performance but I see that these variables are related to air mass flow rates. Are these variables also changing at these times? I think you will have to dig a little deeper into the model to understand if the Heat Pump Electricity Rate shown above is correct or not.

    // OU fan power
    N_fan_OU = N_fan_OU_evap + N_fan_OU_cond;

        N_fan_OU_evap = this->RatedOUFanPower * m_air_evap / m_air_rated;
        N_fan_OU_cond = this->RatedOUFanPower * m_air_cond / m_air_rated;

Reply:
Oh, and add outdoor temperature since that is also a driving force of performance.

Reply:
Also add VRF Heat Pump Outdoor Unit Evaporator Heat Extract Rate so we can see what is happening at the evaporator.

Reply:
@dareumnam and @rraustad. I created the following plots. There is not an output variable for refrigerant flow. I will add some custom outputs later. The Outdoor Unit Evaporator Heat Extract Rate is 0 during cooling mode, in both periods, 04/06 and 04/09.

The following are a few plots for the 04/09 period. The fact that heat pump electricity is higher when cycling ratio < 1 than when cycling ratio = 1, seems to originate from the compressor power, which also has this behavior. I will dig deeper into this.

![image](https://github.com/NREL/EnergyPlus/assets/13421161/a8e00572-81b7-41ad-b97a-7d27589e45d0)
[compare_outputs_cyclingRatio_non1to2_update.xlsx](https://github.com/user-attachments/files/15778642/compare_outputs_cyclingRatio_non1to2_update.xlsx)

Reply:
I looked at the following two points, at 4/6 12:20 and 4/6/ 14:20. The former has heat pump cycling ratio = 0.949, the latter has cycling ratio of 1.0. The former has higher compressor power (which leads to higher heat pump electricity rate) than the latter. 
![image](https://github.com/NREL/EnergyPlus/assets/13421161/3262bd21-9ce7-45a3-aff0-81513a062cf3)

This is because its evaporating temperature is lower. The condensing temperature is pretty similar, lower evaporating temperature will lead to higher curve value (see the following compressor curve at this speed). 

![image](https://github.com/NREL/EnergyPlus/assets/13421161/2fcf98db-a453-4b6e-9e1f-61d54f6c6301)

As the compressor electricity rate is computed as the following, the compressor power (before multiplying cycling ratio is higher) at the 12:20 due to lower evaporating temperature.
```
Ncomp = this->RatedCompPower * CurveValue(state, this->OUCoolingPWRFT(CounterCompSpdTemp), T_discharge, T_suction);
```

The evaporating temperature is lowered during cycling as there was a low-load modification mechanism in the fluid control VRF, where they enlarge the compressor evaporating capacity by changing the evaporating temperature (see the red shaded part), so that the equipment can cycle less.
![image](https://github.com/NREL/EnergyPlus/assets/13421161/268e4cf5-ac08-4a49-891b-bfd9f0edee99)

@rraustad @dareumnam

Reply:
Are the refrigerant mass flow rates the same at both points? (4/6 12:20 and 4/6/ 14:20)

Reply:
Hi @dareumnam, there is not a refrigerant flow rate output variable. I used some debug prints to output this variable (`m_ref_IU_evap_i ` and `m_ref_IU_evap` in cooling mode). The refrigerant flow at 12:20 is 0.01058698505294939, and the refrigerant flow at 14:20 is 0.012904189061317716.